fixture-monkey/src/main/java/com/navercorp/fixturemonkey/expression/StrictModeNextNodePredicateContainer.java

```diff
@@ -61,31 +64,46 @@ public int size() {
 	 * but it does not verify if index {@code 1} is valid for that list.
 	 * This can lead to runtime errors in some edge cases, instead of errors at the validation stage.
 	 *
-	 * @param rootClass            The class to start path validation from.
+	 * @param rootAnnotatedType    The class to start path validation from.
 	 * @param predicates           A list of predicates representing the expression path.
 	 * @param propertyNameResolver The resolver used to check property names.
 	 * @return {@code true} if the path is valid, {@code false} otherwise.
 	 */
 	private boolean isValidFieldPath(
-		Class<?> rootClass, List<NextNodePredicate> predicates, PropertyNameResolver propertyNameResolver
+		AnnotatedType rootAnnotatedType, List<NextNodePredicate> predicates, PropertyNameResolver propertyNameResolver
 	) {
 		if (predicates == null || predicates.isEmpty()) {
 			return false;
 		}
-		Class<?> currentClass = Types.getActualType(rootClass);
 		for (NextNodePredicate predicate : predicates) {
-			if (!(predicate instanceof PropertyNameNodePredicate)) {
+			if (predicate instanceof StartNodePredicate) {
 				continue;
 			}
-			String fieldName = ((PropertyNameNodePredicate)predicate).getPropertyName();
-			Optional<Field> field = Arrays.stream(currentClass.getDeclaredFields())
-				.filter(f -> propertyNameResolver.resolve(new FieldProperty(f)).equals(fieldName))
-				.findFirst();
+			if (predicate instanceof PropertyNameNodePredicate) {
+				String fieldName = ((PropertyNameNodePredicate)predicate).getPropertyName();
+				Class<?> actualClass = Types.getActualType(rootAnnotatedType);
+				Optional<Field> field = Arrays.stream(actualClass.getDeclaredFields())
+					.filter(f -> {
+						String resolvedFieldName = propertyNameResolver.resolve(new FieldProperty(f));
+						return resolvedFieldName.equals(fieldName);
+					})
+					.findFirst();
 
-			if (!field.isPresent()) {
+				if (!field.isPresent()) {
+					return false;
+				}
+
+				rootAnnotatedType = field.get().getAnnotatedType();
+			} else if (predicate instanceof ContainerElementPredicate) {
```

Comment:
It would be better to add the handling logic for the following predicates: `NodeValuePredicate`, `NodeKeyPredicate` and `NodeAllElementPredicate`.

The tests for the expressions `*` and `InnerSpec` refer to `Map` respectively.